GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -125,91 +237,191 @@ static GVRAssimpImporter readFileFromResources(GVRContext gvrContext,
      */
-    static GVRAssimpImporter readFileFromSDCard(GVRContext gvrContext,
+    GVRAssimpImporter readFileFromSDCard(GVRContext gvrContext,
             String filename, EnumSet<GVRImportSettings> settings) {
         long nativeValue = NativeImporter.readFileFromSDCard(filename, GVRImportSettings.getAssimpImportFlags(settings));
         return new GVRAssimpImporter(gvrContext, nativeValue);
     }
 
-    static GVRModelSceneObject loadJassimpModel(final GVRContext context,
-            String filePath, GVRResourceVolume.VolumeType volumeType,
-            EnumSet<GVRImportSettings> settings) throws IOException {
-        return loadJassimpModel(context, filePath, volumeType, settings, false);
+    /**
+     * Loads a scene object {@link GVRModelSceneObject} from
+     * a 3D model and adds it to the scene.
+     *
+     * @param assetFile
+     *            A filename, relative to the root of the volume.
+     *
+     * @param volumeType
+     *            Where the asset is located (SD card, assets directory or network)
+     *            
+     * @param scene
+     *            If present, this asset loader will wait until all of the textures have been
+     *            loaded and then adds the model to the scene.
+     *            
+     * @return A {@link GVRModelSceneObject} that contains the meshes with textures and bones
+     * and animations.
+     * @throws IOException 
+     *
+     */
+    public GVRModelSceneObject loadModel(String filePath, GVRResourceVolume.VolumeType volumeType,
+            GVRScene scene) throws IOException
+    {
+        if (scene != null)
+        {
+            AssetEventListener assetListener = new AssetEventListener(this, scene);
+            getEventReceiver().addListener(assetListener);
+            assetListener.beginLoadAsset(filePath, null);
+        }
+        return loadJassimpModel(filePath, volumeType, GVRImportSettings.getRecommendedSettings(), false);
+    }
+    
+    /**
+     * Loads a scene object {@link GVRModelSceneObject} from
+     * a 3D model and adds it to the scene.
+     *
+     * @param assetFile
+     *            A filename, relative to the root of the volume.
+     *
+     * @param volumeType
+     *            Where the asset is located (SD card, assets directory or network)
+     *            
+     * @param settings
+     *            Additional import {@link GVRImportSettings settings}
+     *
+     * @param cacheEnabled
+     *            If true, add the loaded model to the in-memory cache.
+     *
+     * @param scene
+     *            If present, this asset loader will wait until all of the textures have been
+     *            loaded and then adds the model to the scene.
+     *            
+     * @return A {@link GVRModelSceneObject} that contains the meshes with textures and bones
+     * and animations.
+     * @throws IOException 
+     *
+     */
+    public GVRModelSceneObject loadModel(String filePath, GVRResourceVolume.VolumeType volumeType,
```

Comment:
This is kind of cosmetic but I'd say the final API will be cleaner if we just expose loadModelFromUrl, loadModelFromAssets, loadModelFromSdcard, etc instead of passing the volumeType. It would make for easier to read code and possibly make it a bit less error-prone. Also we might request URL instance from the user in the case of loadModelFromUrl so we know we get valid input, etc.

--
For example the existing API would return a null model instance in case of an invalid URL. The user needs to know what to look for in logcat to find out that.